### ISSUE TITLE
macOS: libusb is not included, that dfu-util depends on

### ISSUE THREAD

**Issue by toyoshim** [rogerclarkmelbourne/Arduino_STM32#256]
https://github.com/rogerclarkmelbourne/Arduino_STM32/wiki/Installation

The document says now that dfu-util is included, users do not need to install it manually.
Actually, a binary is included, but it seems users still need to install libusb on which the distributed binary depends. Otherwise, it just aborts due to missing dylib on starting upload.

**Comment by rogerclarkmelbourne** [comment]
What would you recommend to fix this ?

**Comment by toyoshim** [comment]
Is it easy to build another dfu-util with a static library version of libusb and to push it into the repo.?

**Comment by rogerclarkmelbourne** [comment]
I don't know

You could give it a try and make a PR with the updated version

**Comment by satoshinm** [comment]
I had the same problem, not sure of a proper fix but this seems to work as a temporary workaround:

```sh
brew install libusb
sudo mkdir -p /opt/local/lib
sudo ln -s /usr/local/Cellar/libusb/1.0.21/lib/libusb-1.0.0.dylib /opt/local/lib/libusb-1.0.0.dylib
```

**Comment by rogerclarkmelbourne** [comment]
@satoshinm 

Just out of curiosity, which version of MacOS are you using ?

**Comment by satoshinm** [comment]
Just updated to the latest, macOS High Sierra 10.13.2 public beta 6 (but not sure if using the beta is significant)

**Comment by JBerg60** [comment]
Same error here. Solution was to install libusb `brew install libusb`

**Comment by Cipher2600** [comment]
I am getting the following error upon attempting to compile:

fork/exec /bin/arm-none-eabi-g++: no such file or directory
Error compiling for board Generic STM32F103C series.

I have done everything as stated and also built the libusb. I am new to this so any help would be great! I am running Sierra on a MacBook Pro.

**Comment by ratkins** [comment]
Same issue for me on Mojave (10.14.3 (18D109)). `brew install libusb` (no other `ln` commands necessary) fixes it.

**Comment by stevstrong** [comment]
Is this solved? Possible duplicate of #328 
Anyway, I have no Mac, so any help on this is appreciated.

**Comment by rogerclarkmelbourne** [comment]
This issue has no activity for a long time , so I will close it
